### PR TITLE
Speculative local worskspace (and user) manifest

### PR DESCRIPTION
**Pull request by touilleMan**
replace #1801 with a better approach: the famous `speculative` flag \o/

### CODE REVIEW

File: parsec/core/types/manifest.py [hunk 1]
```diff
@@ -362,17 +364,17 @@ def new_placeholder(
...
     ) -> "LocalFileManifest":
+        id = EntryID.new()
         now = now or pendulum_now()
         blocks = ()
         return cls(
             base=RemoteFileManifest(
                 author=author,
                 timestamp=now,
-                id=id or EntryID.new(),
+                id=id,
```
Comment (vxgmichel) [inline parsec/core/types/manifest.py:377]: :+1:

**Review by vxgmichel — APPROVED** [review]
A few comments, LGTM otherwise :+1:

File: parsec/core/fs/userfs/userfs.py [hunk 2]
```diff
@@ -189,7 +192,7 @@ def __init__(
         self._workspace_storage_nursery: trio.Nursery  # Setup by UserStorage.run factory
         self._process_messages_lock = trio.Lock()
         self._update_user_manifest_lock = trio.Lock()
-        self._workspace_storages: Dict[EntryID, WorkspaceFS] = {}
+        self._workspace_fss: Dict[EntryID, WorkspaceFS] = {}
```
[the remark this answers is not in the document]
Comment (vxgmichel) [reply parsec/core/fs/userfs/userfs.py:195]: Or even `_workspaces` as the instances of `WorkspaceFS` are the closest thing we have to the concept of workspace.

[the remark this answers is not in the document]
Comment (touilleMan) [reply parsec/core/fs/userfs/userfs.py:195]: image

File: parsec/core/fs/workspacefs/sync_transactions.py [hunk 3]
```diff
@@ -177,27 +177,70 @@ def merge_manifests(
     if local_manifest.match_remote(remote_manifest):
         return local_from_remote
 
-    # The remote changes are ours, simply acknowledge them and keep our local changes
-    if remote_manifest.author == local_author:
+    speculative = isinstance(local_manifest, LocalWorkspaceManifest) and local_manifest.speculative
```
[the remark this answers is not in the document]
Comment (touilleMan) [reply parsec/core/fs/workspacefs/sync_transactions.py:180]: :+1:

**Comment by vxgmichel** [comment]
@touilleMan I just rebased this PR on master, feel free to have a look and to sign your commits :)

**Pull request merged**